discovery/gossiper.go

```diff
@@ -1221,9 +1275,25 @@ func (d *AuthenticatedGossiper) processNetworkAnnouncement(nMsg *networkMsg) []l
 			return nil
 		}
 
-		if err := d.waitingProofs.Remove(proof.OppositeKey()); err != nil {
-			err := errors.Errorf("unable remove opposite proof "+
-				"for the channel with chanID=%v: %v", msg.ChannelID, err)
+		// Now set the waitingproof completed, such that it can be
+		// removed from the database if it was sent to the peer.
+		err = d.waitingProofs.SetCompleted(msg.ShortChannelID, true)
```

Comment:
It seems like there's a possible race between this, and the goroutine launched below. This has the potential to cause the waiting proofs to stick around on disk longer than is strictly required. If we crash _before_ we get to this point, then we never mark the proof as completed. As a result, it will never be deleted from disk. 

A remedy to this would be to scan all waiting proofs on startup and delete the iff, if we read the channel edge from disk, it's announcement is already set. I think this works?